rocAL/rocAL/source/coco_meta_data_reader.cpp

```diff
@@ -237,27 +304,39 @@ void COCOMetaDataReader::read_all(const std::string &path)
                 sprintf(buffer, "%012d", id);
                 string str(buffer);
                 std::string file_name = str + ".jpg";
-
                 auto it = _map_img_sizes.find(file_name);
                 ImgSize image_size = it->second; //Normalizing the co-ordinates & convert to "ltrb" format
                 box.l = bbox[0] / image_size.w;
                 box.t = bbox[1] / image_size.h;
                 box.r = (bbox[0] + bbox[2]) / image_size.w;
                 box.b = (bbox[1] + bbox[3]) / image_size.h;
-                bb_coords.push_back(box);
-                bb_labels.push_back(label);
-                add(file_name, bb_coords, bb_labels, image_size);
-                bb_coords.clear();
-                bb_labels.clear();
-                image_size = {};
+                if (_mask && iscrowd == 0)
+                {
+                    bb_coords.push_back(box);
+                    bb_labels.push_back(label);
+                    mask_cords.push_back(mask_cord);
+                    add(file_name, bb_coords, bb_labels, image_size, mask_cords);
+                    mask_cord.clear();
+                    mask_cords.clear();
+                    bb_coords.clear();
+                    bb_labels.clear();
+                }
+                else if (!_mask)
+                {
+                    bb_coords.push_back(box);
+                    bb_labels.push_back(label);
+                    add(file_name, bb_coords, bb_labels, image_size);
+                    bb_coords.clear();
+                    bb_labels.clear();
+                }
             }
         }
         else
         {
             parser.SkipValue();
         }
     }
-    for (auto &elem : _map_content)
+    /*for (auto &elem : _map_content)
```

Comment:
Why was this part commented out?
This part is necessary for SSD pipeline to convert label_id to a continous_label_id (inv_map)

Reply:
Resolved in latest commit